src/saltext/azurerm/modules/azurerm_dns.py

```diff
@@ -340,20 +440,31 @@ def zone_create_or_update(name, resource_group, **kwargs):
         ]
 
     try:
-        zone_model = saltext.azurerm.utils.azurerm.create_object_model("dns", "Zone", **kwargs)
+        zone_model = saltext.azurerm.utils.azurerm.create_object_model(client, obj, **kwargs)
     except TypeError as exc:
         result = {"error": "The object model could not be built. ({})".format(str(exc))}
         return result
 
     try:
-        zone = dnsconn.zones.create_or_update(
-            zone_name=name,
-            resource_group_name=resource_group,
-            parameters=zone_model,
-            if_match=kwargs.get("if_match"),
-            if_none_match=kwargs.get("if_none_match"),
-        )
-        result = zone.as_dict()
+        if zone_type.lower() == "private":
+            zone = dnsconn.private_zones.begin_create_or_update(
+                private_zone_name=name,
+                resource_group_name=resource_group,
+                parameters=zone_model,
+                if_match=kwargs.get("if_match"),
+                if_none_match=kwargs.get("if_none_match"),
+                polling=True,
+            )
+            result = zone.polling_method().resource().as_dict()
```

Comment:
**question (blocking):** The poller doesn't seem to have changed since the "olden days". It has a `wait()` method to ensure the operation is finished, and then the result can be pulled and converted to a dict. It seems like that would be safer than this one-liner. Can you confirm? If so, there are examples in other parts of the extension.